lib/Maintenance/Tasks/LogArchiveTask.php

```diff
@@ -98,7 +98,20 @@ public function execute()
 
             $fileObjectPaths = $db->fetchAllAssociative(sprintf($sql, 'fileobject'));
             foreach ($fileObjectPaths as $objectPath) {
-                $storage->delete($objectPath['fileobject']);
+
+                $filePath = $objectPath['fileobject'];
+                if ($storage->fileExists($filePath)) {
+                    $storage->delete($filePath);
+                } else {
+
+                    // Fallback, if is not found and deleted in the flysystem, tries to delete from local
+                    $fileRealPath = realpath($filePath);
+                    if (str_starts_with(realpath($fileRealPath), PIMCORE_LOG_FILEOBJECT_DIRECTORY)) {
+                        @unlink($fileRealPath);
+                    }
+
+                }
```

Comment:
```suggestion
                if($filePath !== null) {
                    if ($storage->fileExists($filePath)) {
                        $storage->delete($filePath);
                    } else {
    
                        // Fallback, if is not found and deleted in the flysystem, tries to delete from local
                        $fileRealPath = realpath($filePath);
                        if (str_starts_with(realpath($fileRealPath), PIMCORE_LOG_FILEOBJECT_DIRECTORY)) {
                            @unlink($fileRealPath);
                        }  
                    }
                 }
```

`$storage->fileExists` will fail if `$filePath` is null